### PR TITLE
ci: adding github actions to create nightly builds

### PR DESCRIPTION
**Pull request by marosset**
**Reason for Change**:

Add CI for adding a nightly build/release of aks-engine.
This will help increase test velocity for various upstream projects that rely on aks-engine for testing.

**Issue Fixed**:

**Credit Where Due**:

Does this change contain code from or inspired by another project?

- [x] No
- [ ] Yes

If "Yes," did you notify that project's maintainers and provide attribution?

- [ ] No
- [ ] Yes

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Review by chewong — APPROVED** [review]

File: .github/workflows/nightly-build.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+jobs:
+
+  build:
+    name: build
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (chewong) [inline .github/workflows/nightly-build.yaml:16]: does it always check out the master branch?

Comment (marosset) [reply .github/workflows/nightly-build.yaml:18]: Whoops, i removed the branch specification when i removed the pull-request trigger i was using for testing.

Comment (marosset) [reply .github/workflows/nightly-build.yaml:16]: Yea, the schedule trigger works on the default/master branch.

Here is a discussion i found about this.
https://github.community/t/on-schedule-per-branch/17525

Comment (marosset) [reply .github/workflows/nightly-build.yaml:16]: actions/checkout will checkout whatever ref triggered the action (you can override it tho).
The on->schedule action currently only triggers against the default branch in the repo.

**Comment by marosset** [comment]
Admin merging since github <-> azure devops pipeline integration is flakey today and this PR does not touch any code.

**Pull request merged**